Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -94,6 +96,38 @@ public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskC
 
         #endregion
 
+        /// <inheritdoc/>
+        public Task<OIdCToken> GetOIdCTokenAsync(OIdCTokenOptions options, CancellationToken cancellationToken = default)
+        {
+            options.ThrowIfNull(nameof(options));
+            // If at some point some properties are added to OIdCToken that depend on the token having been fetched
+            // then initialize the token here.
+            TokenRefreshManager tokenRefreshManager = null;
```

Comment:
This means if we call this method multiple times, we're going to fetch it multiple times... would it be worth having a lazily-instantiated TokenRefreshManager for OIDC tokens from this credential as an instance field, so we get proper caching?

Reply:
I thought about caching it, and I still can do it, but we would have to keep a cache per `OIdCTokenOptions`. The `TokenRefreshManager` and not the `OIdCToken` it's the one that keeps a reference to the `TokenResponse` and checks if it has expired before requesting another. In turn the `OIdcToken` only holds a reference to its own `TokenRefreshManager` That's weird, but it was done like that because of the public surface of the existing credential types, there's a note at the beginning of the `TokenRefreshManager` class even. I'm just piggy backing on that. But happy to do a cache on options really.
(BTW, if the problem is that this is capturing the wrong things, I knew that might be so, I was just waiting for tests to fix it if needed.)